doc/sphinx/language/gallina-specification-language.rst

```diff
@@ -614,148 +619,101 @@ syntax: :n:`let fix @ident @binders := @term in` stands for
-                      : Let `ident` [`binders`] [: `term`] := `term` .
-   binders            : binders binder
-                      : binder
-   inductive          : Inductive `ind_body` with … with `ind_body` .
-                      : CoInductive `ind_body` with … with `ind_body` .
-   ind_body           : `ident` [`binders`] : `term` :=
-                      : [[|] `ident` [`binders`] [:`term`] | … | `ident` [`binders`] [:`term`]]
-   fixpoint           : Fixpoint `fix_body` with … with `fix_body` .
-                      : CoFixpoint `fix_body` with … with `fix_body` .
-   assertion          : `assertion_keyword` `ident` [`binders`] : `term` .
-   assertion_keyword  : Theorem | Lemma
-                      : Remark | Fact
-                      : Corollary | Property | Proposition
-                      : Definition | Example
-   proof              : Proof . … Qed .
-                      : Proof . … Defined .
-                      : Proof . … Admitted .
-   decoration : #[ `attributes` ]
-   attributes : [`attribute`, … , `attribute`]
-   attribute :  `ident`
-             : `ident` = `string`
-             : `ident` ( `attributes` )
-
-.. todo:: This use of … in this grammar is inconsistent
-          What about removing the proof part of this grammar from this chapter
-          and putting it somewhere where top-level tactics can be described as well?
-          See also #7583.
+.. insertprodn vernacular vernacular
+
+.. prodn::
+   vernacular ::= {* {? @all_attrs } {| @command | @ltac_expr } . }
 
-This grammar describes *The Vernacular* which is the language of
-commands of Gallina. A sentence of the vernacular language, like in
-many natural languages, begins with a capital letter and ends with a
-dot.
+The top-level input to |Coq| is a series of commands and tactics, each terminated with a period
+and optionally decorated with :ref:`gallina-attributes`.  :n:`@ltac_expr` syntax supports both simple
+and compound tactics.  For example: ``split.`` is a simple tactic while ``split; auto.`` combines two
+simple tactics.
 
-Sentences may be *decorated* with so-called *attributes*,
-which are described in the corresponding section (:ref:`gallina-attributes`).
+Tactics specify how to transform the current proof state in order to generate a proof.  They
+are syntactically valid only when |Coq| is in proof mode, such as after a :cmd:`Theorem` command
+and before any subsequent proof-terminating command such as :cmd:`Qed`.  See :ref:`proofhandling` for more
+on proof mode.
 
-The different kinds of command are described hereafter. They all suppose
-that the terms occurring in the sentences are well-typed.
+By convention, :n:`@command` names begin with uppercase letters, while :n:`@tactic` names begin with lowercase letters.
+
+.. todo:: What about removing the proof part of this grammar from this chapter
+          and putting it somewhere where top-level tactics can be described as well?
+          See also #7583.
```

Comment:
I love what you wrote in this introduction. I feel like the small look-ahead to the chapters about proof modes, tactics and tactic language is fine, but we could also do without. For now, I would keep this as is.